BEDS-839/remove timestamp pending and exiting validators

enzo-bitfly opened the pull request:
**Changes besides ticket description:**
- fix dependency vulnerabilities
- use api-provided `PostValidatorDashboardValidatorsRequest` for validator dashboard `update` api call - api added this type recently and we needed to use it in the frontend instead of our hardcoded type

enzo-bitfly committed:
fix(DashboardValidatorSubsetList): remove go timestamp from `pending` and `exiting` validators

- add missing relative timestamp to `exiting` validator
- the api uses the `duty_object` property for the timestamp of `pending` and `exiting` validators, which is incorrect, so we need to handle their case to create a relative timestamp as label, and prevent creating links, which are meant for duty-related data

enzo-bitfly committed:
refactor(DashboardValidatorManagmentModal): replace hardcoded body type for validator dashboard post request by api-provided

marcel-bitfly commented on frontend/components/dashboard/validator/subset/DashboardValidatorSubsetList.vue:
```diff
@@ -105,12 +105,16 @@ function mapDutyLinks(
       break
   }
+
   if (path) {
     return dutyObjects.map(o => ({
       label: `${formatValue ? formatNumber(o) : o}`,
       to: `${path}${o}`,
     }))
   }
+  else if (props.category === 'pending' || props.category === 'exiting') {
+    return []
+  }
```
would this also work?
```suggestion
  if (props.category === 'pending' || props.category === 'exiting') return []
 
  if (path) {
    return dutyObjects.map(o => ({
      label: `${formatValue ? formatNumber(o) : o}`,
      to: `${path}${o}`,
    }))
  }
```

enzo-bitfly replied:
Good catch, I applied your suggestion

marcel-bitfly commented on frontend/components/dashboard/validator/subset/DashboardValidatorSubsetList.vue:
```diff
@@ -139,9 +143,9 @@ function mapDutyLinks(
+            <span class="label">{{ createDutyLabels(v.duty_objects) }}</span>
             <template
-              v-for="link in mapDutyLinks(v.duty_objects)"
+              v-for="link in createDutyLinksData(v.duty_objects)"
```
would this make sense?
```suggestion
              v-for="link in dutylinks(v.duty_objects)"
```

enzo-bitfly replied:
It's a function so the name is an action + high context + low context. Also, it returns link data, not a link.

marcel-bitfly commented on frontend/components/dashboard/validator/subset/DashboardValidatorSubsetList.vue:
```diff
@@ -18,7 +18,6 @@ const { t: $t } = useTranslation()
 const paging = ref<null | Paging>(null)
 const cursor = ref<Cursor>(undefined)
 const VALIDATORS_PER_PAGE = 100
-
```
np: ideally we are very rigorous with touching `whitespace`, as it will make it `line blaming` a lot harder the overtime

enzo-bitfly replied:
That was an accident, I reverted it

marcel-bitfly commented on frontend/nuxt.config.ts:
```diff
@@ -122,5 +118,6 @@ export default defineNuxtConfig({
   },
   postcss: { plugins: { autoprefixer: {} } },
   eslint: { config: { stylistic: true } },
+
```
np: whitespace

enzo-bitfly replied:
Don't know what happened there. Fixed

enzo-bitfly committed:
chore(deps): resolve security vulnerabilities in dependencies

- remove unnecessary `commonjs` plugin from vite config, which broke the build after this update
- vulnerabilities fixed:
  https://github.com/advisories/GHSA-vg6x-rcgg-rjx6
  https://github.com/advisories/GHSA-2452-6xj8-jh47
